### ISSUE TITLE
`wrangler dev` fails when `workers_dev = true` and custom routes are specified

### ISSUE THREAD

**Issue by npajkovsky** [cloudflare/workers-sdk#2031]
#### Reproduce the bug

It's very simple. I have a rust project, and I have run this with `wrangler.toml`.

```
name = "do-ws"
workers_dev = true

main = "build/worker/shim.mjs"
compatibility_date = "2022-01-20"

routes = [
    { pattern = "u-tygra-dev.api.pajkovsky.io", custom_domain = true, zone_id = "..." }
]

[build]
command = "cargo install -q worker-build --version 0.0.7 && worker-build --release"

[durable_objects]
bindings = [
  { name = "WEBSOCKETS", class_name = "WebSocketStore" }
]

[[migrations]]
tag = "v1"
new_classes = ["WebSocketStore"]

[env.prod]
name = "do-ws-prod"
workers_dev = false
routes = [
    { pattern = "u-tygra.api.pajkovsky.io", custom_domain = true, zone_id = "..." }
]
vars = { ENVIRONMENT = "prod" }
durable_objects.bindings = [
  {name = "WEBSOCKETS", class_name = "WebSocketStore", script_name = "do-ws-prod"}
]
```
And running `wrangler dev` fails with error

```
- http://192.168.1.244:8787
✘ [ERROR] Error while creating remote dev session: TypeError: fetch failed

      at Object.processResponse
  (/Users/npajkovsky/freelancer/u-tygra/be/node_modules/wrangler/wrangler-dist/cli.js:8846:27)
      at /Users/npajkovsky/freelancer/u-tygra/be/node_modules/wrangler/wrangler-dist/cli.js:9178:42
      at node:internal/process/task_queues:141:7
      at AsyncResource.runInAsyncScope (node:async_hooks:202:9)
      at AsyncResource.runMicrotask (node:internal/process/task_queues:138:8)
      at processTicksAndRejections (node:internal/process/task_queues:96:5) {
    cause: Error: getaddrinfo ENOTFOUND pajkovsky.io
        at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
      errno: -3008,
      code: 'ENOTFOUND',
      syscall: 'getaddrinfo',
      hostname: 'pajkovsky.io'
    }
  }
```

#### Environment and versions

* operating system: Linux/Mac
* output of `wrangler -V`:
```
 ⛅️ wrangler 2.1.11
--------------------
```
* output of `node -v`:
v16.16.0

Why am I getting this error?

**Comment by caass** [comment]
Hmm...it looks like there's a DNS error? In the `ENOTFOUND` error:

```text
    cause: Error: getaddrinfo ENOTFOUND pajkovsky.io
        at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
      errno: -3008,
      code: 'ENOTFOUND',
      syscall: 'getaddrinfo',
      hostname: 'pajkovsky.io'
    }
```

Just quickly running `whois` in the terminal, I can see the domain is on cloudflare, and while DNS lookups to `pajkovsky.io` fail, I'm not sure that that should be an issue. I think there had been an issue around using `wrangler dev` with durable objects where the worker hadn't been published before, but I can't remember, really.

Have you published this worker before? With the durable object binding? Also, does wrangler work as expected when running in local mode (`wrangler dev -l`)?

**Comment by npajkovsky** [comment]
Looks like it is related to `routes`. When I have commented out the dev route, it works like a charm.

**Comment by caass** [comment]
Oh interesting...hmm. We'll investigate and get back to you :)

**Comment by ahaoboy** [comment]
Same issue.
  
System:
    OS: macOS 13.0
  Binaries:
    Node: 18.9.0 - ~/Library/Caches/fnm_multishells/60050_1667356309613/bin/node
    Yarn: 1.22.19 - ~/Library/Caches/fnm_multishells/60050_1667356309613/bin/yarn
    npm: 8.19.1 - ~/Library/Caches/fnm_multishells/60050_1667356309613/bin/npm
  Browsers:
    Chrome: 107.0.5304.87
    Edge: Not Found
    Firefox: Not Found
    Safari: 16.1

**Comment by andrewcoelho** [comment]
I'm getting this error with the default template after running `wrangler init` and then `wrangler dev`

**Comment by penalosa** [comment]
I believe this is because we make some requests to the route zone (prewarm, I think, and possibly others). A quick win could be to find those requests and make them to the custom route instead.

**Comment by jspspike** [comment]
This was fixed in #2651, the host domain for the dev session will be used for the exchange. This will either be the first route specified or explicitly with the [`host` field](https://developers.cloudflare.com/workers/wrangler/configuration/#local-development-settings).
